common/djangoapps/student/views.py

```diff
@@ -1599,6 +1617,31 @@ def create_account_with_params(request, params):
         getattr(settings, 'REGISTRATION_EXTRA_FIELDS', {})
     )
 
+    # Start: Added by Labster
+    # User can create an account only in the appropriate region or user
+    # that already enrolled in this current region.
+    email = params["email"]
+    email_enrolled = CourseEnrollmentAllowed.objects.filter(email=email).exists()
+    if not email_enrolled:
+        regions = configuration_helpers.get_value('REGIONS', settings.REGIONS)
+        region = get_user_region(request, regions, email, enrollment=True)
+        # Check if user already enrolled in other regions.
+        if region:
+            msg = _(
+                'It appears that you are already enrolled in a course hosted on our {region_code} server. '
+                'Please sign up <a href="{register_url}">here</a> instead.'
+            ).format(
+                region_code=region['region_code'],
+                register_url=region['register_url'],
+            )
+            raise ValidationError({'error': [msg]})
+
+        ip_address_warning_msg = get_ip_address_warning_msg(request, regions)
+        if ip_address_warning_msg:
```

Comment:
I realize in our diagram we don't have any warning message if there's no result from checking enrollment in list regions, I already talk to @avkoval , he will double check with Emma about this so for temporary we return IP check again to prevent user signup based on IP address.

Reply:
Please add some flag to be able to manage whether we want to allow individual users from the other regions to sign up.